src/cache/gcs.rs

```diff
@@ -13,51 +13,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::{fmt, io, sync::Arc, time};
 use crate::{
     cache::{Cache, CacheRead, CacheWrite, Storage},
     errors::*,
     util::HeadersExt,
 };
-use futures_03::future::Shared;
 use hyper::Method;
 use hyperx::header::{Authorization, Bearer, ContentLength, ContentType};
 use reqwest::{Client, Request};
 use serde::de;
-use std::{convert::Infallible, sync};
-use std::{fmt, io, pin::Pin, result, sync::Arc, time};
 use url::{
     form_urlencoded,
     percent_encoding::{percent_encode, PATH_SEGMENT_ENCODE_SET, QUERY_ENCODE_SET},
 };
-// use ::ReqwestRequestBuilderExt;
-use futures_03::FutureExt;
 
-#[derive(thiserror::Error, Debug, Clone)]
-pub enum Error {
-    #[error("Http error: {0}")]
-    Http(#[from] crate::errors::BadHttpStatusError),
-
-    #[error("Error: {0}")]
-    Arbitrary(String),
-}
-
-impl From<String> for Error {
-    fn from(s: String) -> Self {
-        Self::Arbitrary(s.to_string())
-    }
-}
-
-impl From<&str> for Error {
-    fn from(s: &str) -> Self {
-        Self::Arbitrary(s.to_owned())
-    }
-}
-
-impl From<reqwest::Error> for Error {
-    fn from(s: reqwest::Error) -> Self {
-        Self::Arbitrary(s.to_string())
-    }
-}
```

Comment:
This isn't used upstream and I can't seem to find other module-specific Errors other than 
https://github.com/mozilla/sccache/blob/b3f4727eda81c45b0632d036395af85429c4774b/src/compiler/args.rs#L39
https://github.com/mozilla/sccache/blob/385f738c5281af2ebcd00d9f46e37424ae16ea39/src/lru_disk_cache/mod.rs#L66

Since usage of `anyhow` is so pervasive, could we omit this new `Error` from the diff? We don't seem to be handling it and it looks like it's either `BadHttpStatusError` or an inner error converted to `String`, which looks to me more like a use case for a regular `anyhow` error flow.